### PR TITLE
[Driver][SYCL] Fix integration header handling with preprocessed files

### PR DESCRIPTION
**Pull request by mdtoguchi**
When generating preprocessed files using -fsycl, The preprocessed
files did not contain the integration header causing the host side
of the preprocessed file to not contain that information.  Compiling
from preprocessed files are not expected to do another preprocessing
step so the addition of the integration header is lost.

### CODE REVIEW

File: clang/test/Driver/sycl-host-compiler.cpp [hunk 1]
```diff
@@ -27,13 +27,11 @@
 // RUN: %clangxx -fsycl -fsycl-host-compiler=g++ -E %s -### 2>&1 \
 // RUN:  | FileCheck -check-prefix=HOST_PREPROCESS %s
 // HOST_PREPROCESS: g++{{.*}} "-E"{{.*}} "-o" "[[PPOUT:.+\.ii]]"
-// HOST_PREPROCESS: g++{{.*}} "-E"{{.*}} "-o" "-"
```
Comment (AGindinson) [inline clang/test/Driver/sycl-host-compiler.cpp:30]: Is this going to require a rebase on #3645?

File: clang/lib/Driver/Driver.cpp [hunk 2]
```diff
@@ -3854,24 +3854,35 @@ class OffloadingActionBuilder final {
...
+              // Add an additional compile action to generate the integration
+              // header.
+              Action *CompileAction =
+                  C.MakeAction<CompileJobAction>(A, types::TY_Nothing);
+              DA.add(*CompileAction, *ToolChains.front(), nullptr,
+                     Action::OFK_SYCL);
+            } else
+              A = C.getDriver().ConstructPhaseAction(C, Args, CurPhase, A,
+                                                     AssociatedOffloadKind);
+          }
```
Comment (AGindinson) [inline clang/lib/Driver/Driver.cpp:3879]: Would something like this preserve the functional meaning?
The early exit bit feels optional.
```suggestion
            A = C.getDriver().ConstructPhaseAction(C, Args, CurPhase, A,
                                                   AssociatedOffloadKind);
            if (SYCLDeviceOnly)
              continue;
            // Add an additional compile action to generate the integration
            // header.
            Action *CompileAction =
                C.MakeAction<CompileJobAction>(A, types::TY_Nothing);
            DA.add(*CompileAction, *ToolChains.front(), nullptr,
                   Action::OFK_SYCL);
          }
```

File: clang/test/Driver/sycl-host-compiler.cpp (hunk 1 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-host-compiler.cpp:30]: Yes, a rebase is required.  #3645 looks to have been merged, so I will be updating this to reflect.

**Review by AGindinson — APPROVED** [review]
LGTM!

**Pull request merged**